repro_plugin.mjs

```diff
@@ -0,0 +1,33 @@
+import {
+    gql,
+    makeExtendSchemaPlugin,
+} from 'postgraphile/utils'
+import { constant,  } from 'postgraphile/grafast'
+
+
+export const ReproPlugin = makeExtendSchemaPlugin((build) => {
+    const { posts } = build.input.pgRegistry.pgResources
+    const { sql } = build
+    return {
+      typeDefs: gql`
+        extend type Post {
+          repro: Post
+        }
+      `,
+      plans: {
+        Post: {
+          repro($post) {
+            const $posts = posts.find()
+            const postsTbl = $posts.alias
+            $posts.where(sql`
+              ${postsTbl}.id
+                 = ${$post.getClassStep().alias}.id
+            `)
```

Comment:
For future readers, the issue was that an unrelated alias was being referenced (`$post.getClassStep().alias` did not exist in the `$posts.find()` query), and instead a placeholder should be used to allow referencing a value from another query:

```diff
-           const postsTbl = $posts.alias
            $posts.where(sql`
-             ${postsTbl}.id
-                = ${$post.getClassStep().alias}.id
+             ${$posts}.id = ${$posts.placeholder($post.get('id'))}
            `)
```

Note that this may later be optimised so that the two queries become one, but you cannot make that assumption off the bat.